### ISSUE TITLE
Add Covers PHPUnit Annotation to Controller Baked Classes

### ISSUE THREAD

**Issue by justinyost** [loadsys/CakePHP-LoadsysTheme#16]
More information: https://phpunit.de/manual/current/en/appendixes.annotations.html#appendixes.annotations.covers.tables.annotations

Note `@covers` requires the full namespaced path, like so `@covers \App\Controller\Anon\CampSessionsController::index`

**Comment by beporter** [comment]
I'm on the fence about this without going the extra step and having explicit unit vs integration test suites for controllers. As much as I think there are problems with "defaulting" to integration tests, it's what Cake 3 expects, and despite appearances I don't really want to stray _too_ far from that.

In addition, the `@covers` annotation itself is a little dangerous, especially when changing method names. Forgetting to update the annotation means that you get an unexplained drop in coverage. I guess this is mitigated pretty well by ensuring a minimum coverage percentage, but it can be difficult to debug why something you think should be getting covered isn't until you're in the habit of checking for the annotation. This is something we'll adapt to if it's "normal", but it's worth pointing out.

Another (possible) complication is that this gets really sticky when you get into prefixed/namespaced classes. Take a look at what we had to do in Eyesore because of the way we tested each namespaced method using a shared test case using each user role as a `@dataProvider`: https://github.com/loadsys/Eyesore-ORB/pull/25/files#diff-e7e26aeee7bd3d540cc16d44d4affb14R85

**Comment by justinyost** [comment]
Yeah I just updated the comment to include that it would require the full namespaced path.

**Comment by justinyost** [comment]
The big thing this is that trying to get coverage in a project means I have to fight with the integration tests doing coverage because the coverage is lying to me.

**Comment by beporter** [comment]
Yeah, that's a valid point. I thought I had opened a `core` or `bake` issue about that, but I can't seem to find it now.

**Comment by justinyost** [comment]
Regarding the changing method names, that makes sense for things like Tables, Components, etc but since Controller method names are (generally) the url being hit they rarely change at least what we've generally observed. Adding it only to the Controller might be a way of mitigating this need to change the annotation when updating a method name.
